tags-config-service-impl/src/main/java/org/hypertrace/tag/config/service/TagConfigServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+import org.hypertrace.tag.config.service.v1.DeleteTagResponse;
+import org.hypertrace.tag.config.service.v1.GetTagRequest;
+import org.hypertrace.tag.config.service.v1.GetTagResponse;
+import org.hypertrace.tag.config.service.v1.GetTagsRequest;
+import org.hypertrace.tag.config.service.v1.GetTagsResponse;
+import org.hypertrace.tag.config.service.v1.Tag;
+import org.hypertrace.tag.config.service.v1.TagConfigServiceGrpc;
+import org.hypertrace.tag.config.service.v1.UpdateTagRequest;
+import org.hypertrace.tag.config.service.v1.UpdateTagResponse;
+
+@Slf4j
+public class TagConfigServiceImpl extends TagConfigServiceGrpc.TagConfigServiceImplBase {
+  private final ConfigServiceCoordinator configServiceCoordinator;
+
+  public TagConfigServiceImpl(Channel configChannel) {
+    configServiceCoordinator = new ConfigServiceCoordinatorImpl(configChannel);
+  }
+
+  @Override
+  public void createTag(
+      CreateTagRequest request, StreamObserver<CreateTagResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    CreateTag createTag = request.getTag();
+    Tag tag =
+        Tag.newBuilder().setId(UUID.randomUUID().toString()).setKey(createTag.getKey()).build();
+    Tag createdTag = configServiceCoordinator.upsertTag(requestContext, tag);
+    responseObserver.onNext(CreateTagResponse.newBuilder().setTag(createdTag).build());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void getTag(GetTagRequest request, StreamObserver<GetTagResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    String tagId = request.getId();
+    Tag tag = configServiceCoordinator.getTag(requestContext, tagId);
+    responseObserver.onNext(GetTagResponse.newBuilder().setTag(tag).build());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void getTags(GetTagsRequest request, StreamObserver<GetTagsResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    List<Tag> tagList = configServiceCoordinator.getAllTags(requestContext);
+    responseObserver.onNext(GetTagsResponse.newBuilder().addAllTags(tagList).build());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void updateTag(
+      UpdateTagRequest request, StreamObserver<UpdateTagResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    Tag updateTag = request.getTag();
+    Tag updatedTag = configServiceCoordinator.upsertTag(requestContext, updateTag);
+    responseObserver.onNext(UpdateTagResponse.newBuilder().setTag(updatedTag).build());
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void deleteTag(
+      DeleteTagRequest request, StreamObserver<DeleteTagResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    String tagId = request.getId();
+    configServiceCoordinator.deleteTag(requestContext, tagId);
```

Comment:
We should get tag to check if it exists, before delete

Reply:
Yes, I will make this change.

Reply:
Can you check behavior on config service end? Probably it is already handled there itself.

Reply:
If we try to getConfig() on an ID that is not present, it is throwing an exception. So I think all the methods will also throw an exception. We have two options
1. We can add contains(id)/has(id) functionality to config-service.
2. In Tag-Config-Service, we can do a retrieve all the tags and filter them based on ID.
Let me know which approach to take.

Reply:
If getConfig() throws exception, then we just need to ensure it goes all the way back to the caller.

I am assuming deleteConfig() will also throw an exception, and we should ensure it goes all the way to the caller.